src/Viewer.tsx

```diff
@@ -355,6 +358,16 @@ function Viewer(): ReactElement {
   );
 
   // DATASET LOADING ///////////////////////////////////////////////////////
+
+  const handleProgressUpdate = useCallback(
+    (complete: number, total: number): void => {
+      startTransition(() => {
+        setDatasetLoadProgress(Math.round((complete / total) * 100));
+      });
+    },
+    [setDatasetLoadProgress]
+  );
```

Comment:
The progress value is expected to update very quickly and trigger a lot of re-renders, so I've wrapped this in a transition to allow the render to be interrupted when the progress value updates.